pkg/pipelines/service_rolebinding.go

```diff
@@ -0,0 +1,45 @@
+package pipelines
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	v1rbac "k8s.io/api/rbac/v1"
+)
+
+// createServiceAccount creates a ServiceAccount given saName and secretName
+func createServiceAccount(saName, secretName string) corev1.ServiceAccount {
+	return corev1.ServiceAccount{
+		TypeMeta:   createTypeMeta("ServiceAccount", "v1"),
+		ObjectMeta: createObjectMeta(saName),
+		Secrets: []corev1.ObjectReference{
+			corev1.ObjectReference{Name: secretName},
+		},
+	}
+}
+
+// createRoleBinding creates a RoleBinding given name, saName, roleKind, and roleName
+func createRoleBinding(name, saName, roleKind, roleName string) v1rbac.RoleBinding {
```

Comment:
This could take a Role and ServiceAccount?

Reply:
These are eally references (to SA and Role) to build a  RoleBinding return struct.   See below.
```return v1rbac.RoleBinding{
		TypeMeta:   createTypeMeta("RoleBinding", "rbac.authorization.k8s.io/v1"),
		ObjectMeta: createObjectMeta(name),
		Subjects: []v1rbac.Subject{
			v1rbac.Subject{
				Kind: "ServiceAccount",
				Name: saName,
			},
		},
		RoleRef: v1rbac.RoleRef{
			APIGroup: "rbac.authorization.k8s.io",
			Kind:     roleKind,
			Name:     roleName,
		},
```
RoleRef is obviously a reference to a Role by name.   The subject slice is also references to object by name.  It does not have secret or anything like that.   A real ServiceAccount object looks something like this. 

```return corev1.ServiceAccount{
		TypeMeta:   createTypeMeta("ServiceAccount", "v1"),
		ObjectMeta: createObjectMeta(saName),
		Secrets: []corev1.ObjectReference{
			corev1.ObjectReference{Name: secretName},
		},
	}
```

Reply:
It turns out that Role and ClusterRole are the same type.  You can't do it for Role/ClusterRole anyway.